src/AbstractRole.php

```diff
@@ -103,16 +103,33 @@ public function addChild($child)
      */
     public function setParent($parent)
     {
-        $this->parent = $parent;
-
+        if (null === $this->parents) {
+            $this->parents = [];
+        }
+        if (!in_array($parent, $this->parents)) {
+            $this->parents[] = $parent;
+        }
         return $this;
     }
 
     /**
-     * @return null|RoleInterface
+     * @return null|RoleInterface|array
      */
     public function getParent()
     {
-        return $this->parent;
+        if (1 === count($this->parents)) {
+            return $this->parents[0];
+        }
+        return $this->parents;
+    }
+
+    /**
+     * @param  RoleInterface $parent
+     * @return RoleInterface
+     */
+    public function addParent($parent)
```

Comment:
As this isn't being used from `setParent()`, is it necessary anymore?

Reply:
The idea is to leave `setParent()` for BC and add `addParent()` without adding it to `RoleInterface`. In this way we can release 2.6.0 and change the `RoleInterface` until 3.0.0, as commented [here](https://github.com/zendframework/zend-permissions-rbac/pull/23#issuecomment-268006556).